### PR TITLE
Merge local pillars for rsync

### PR DESCRIPTION
**Pull request by brettminnie**
* Can pass a list of search paths for split pillars

### CODE REVIEW

File: cotton/salt/pillar.py [hunk 1]
```diff
@@ -83,30 +101,81 @@ def get_rendered_pillar_location(pillar_dir=None, projects_location=None, parse_
...
-            for file_name in files:
-                files_to_render.append(os.path.join(rel_path, file_name))
+        for pillar in pillars:
+            for root, dirs, files in os.walk(pillar):
+                rel_path = os.path.relpath(root, pillar)
+                for file_name in files:
+                    files_to_render.append(os.path.join(rel_path, file_name))
+
+    if __render_templates(files_to_render, dest_location, jinja_env) is False:
+        print(red("Aborting due to pillar failing to render"))
```
Comment (tim-mcarthur-moj) [inline cotton/salt/pillar.py]: perhaps you could include the failing filename in this error message

Comment (brettminnie) [inline cotton/salt/pillar.py]: Could be list of failures, but will add them

Comment (tim-mcarthur-moj) [inline cotton/salt/pillar.py]: I see that now.  Probably no need if it is a list

Comment (brettminnie) [inline cotton/salt/pillar.py]: @tim-mcarthur-moj  Nope, they where marked above as failures, have added it more explicitly now

Sample output

```
Pillar could not compile the following templates:
 - ./foo.sls
 - ./bar.sls
```

**Pull request merged**